src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -372,6 +441,99 @@ public void appendMetaColIfNotExist(final String tableName) {
     }
   }
 
+  /**
+   * Check whether the user has the role privilge to do schema evolution and whether the schema
+   * evolution option is enabled on the table
+   *
+   * @param tableName the name of the table
+   * @param role the role of the user
+   * @return whether schema evolution has the required permission to be performed
+   */
+  @Override
+  public boolean hasSchemaEvolutionPermission(String tableName, String role) {
+    checkConnection();
+    InternalUtils.assertNotEmpty("tableName", tableName);
+    String query = "show grants on table identifier(?)";
+    ResultSet result = null;
+    boolean hasRolePrivilege = false;
+    boolean hasTableOptionEnabled = true;
+    String myRole = formatRoleName(role);
+    // the schema evolution permission is still in PrPr, so it is left as true here
```

Comment:
any idea about the timeline? And could you create a JIRA dependency and assign to the correct person?